File: GGAndroid_2.0/GGScanner/scanner-sample/src/main/java/io/scanner/sample/MainActivity.java
package io.scanner.sample;

import io.ganguo.utils.util.log.Logger;
import io.ganguo.vmodel.ViewModelActivity;
import io.ganguo.viewmodel.databinding.IncludeHfSwipeRecyclerBinding;
import io.scanner.sample.viewmodel.MainViewModel;

/**
 * <p>
 * 扫码Demo首页
 * </p>
 * Created by leo on 2018/8/9.
 */
public class MainActivity extends ViewModelActivity<IncludeHfSwipeRecyclerBinding, MainViewModel> {
    @Override
    public MainViewModel createViewModel() {
        return new MainViewModel();
    }

    @Override
    public void onViewAttached(MainViewModel viewModel) {
        Logger.init(getContext().getApplicationContext());
    }


}

File: GGAndroid_3.0/main/app-scanner-sample/src/main/java/io/scanner/sample/MainActivity.kt
package io.scanner.sample

import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.databinding.IncludeHfSwipeRecyclerBinding
import io.ganguo.viewmodel.core.view.ViewModelActivity
import io.scanner.sample.viewmodel.MainViewModel


/**
 * <pre>
 * author : leo
 * time   : 2018/8/9
 * desc   : Demo 扫码Demo首页
 * </pre>
 */
class MainActivity : ViewModelActivity<IncludeHfSwipeRecyclerBinding, MainViewModel>() {
    override fun createViewModel(): MainViewModel {
        return MainViewModel()
    }

    override fun onViewAttached(viewModel: BaseViewModel<Any>) {}


}
